locals.tf

```diff
@@ -1,52 +0,0 @@
-locals {
-  default_helm_values = [
-    templatefile("${path.module}/vault-config.tftpl.yml", {})
-  ]
-
-  default_helm_config = {
-    name                       = "vault"
-    chart                      = "vault"
-    repository                 = "https://helm.releases.hashicorp.com"
-    version                    = "0.19.0"
-    namespace                  = "vault"
-    timeout                    = "1200"
-    create_namespace           = true
-    set                        = []
-    set_sensitive              = []
-    lint                       = false
-    values                     = local.default_helm_values
-    wait                       = true
-    wait_for_jobs              = false
-    description                = "Helm chart to install Vault and other associated components"
-    verify                     = false
-    keyring                    = ""
```

Comment:
we can take advantage of things like `lookup()` or `try()` to first attempt to get the value from `var.helm_config` if the user has provided one, otherwise fall back to a default value. When falling back to a default value and we don't have an opinion on an optional argument, we should default to `null` to indicate as such and leave it up to the resource's defaults